### PR TITLE
Adds canister bombs

### PR DESCRIPTION
**Pull request by void5402**
# Document the changes in your pull request

makes canisters explode when they break if they have 200000Kpa or higher pressure and makes single tank bombs attachable to them so the canister can detonate by itself.

# Why is this good for the game?

gives more depth and possibilities to some jobs

# Testing
<details> 
  <summary>tested</summary>
  
[2024-02-29 19-38-18.webm](https://github.com/yogstation13/Yogstation/assets/74709370/db34cc29-3f3c-497e-a4ce-a4e3d5492bf5)

  a tiny bit laggy
</details>

# Wiki Documentation
canisters exploding with high pressure and single tank attachable to them

# Changelog

:cl:  
rscadd: canister bombs now explode when broken if they have enough pressure
rscadd: makes single tank bombs attachable to canisters

/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -33,6 +35,8 @@
 	var/timing = FALSE
 	var/restricted = FALSE
 	req_access = list()
+	//a singletank bomb te explode the canister
+	var/obj/item/tank/single_tank
```
Comment (iamthesilly) [inline code/modules/atmospherics/machinery/portable/canister.dm]: Canisters already have a variable for this, `holding`.

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 2]
```diff
@@ -368,6 +374,29 @@
 
 	return TRUE
 
+//attaching or droping the singletank bomb
+/obj/machinery/portable_atmospherics/canister/attackby(obj/item/W, mob/user, params)
```
Comment (iamthesilly) [inline code/modules/atmospherics/machinery/portable/canister.dm]: Canisters already hold tanks, this entire part is unnecessary.

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/portable/canister.dm (hunk 2 above)
Comment (void5402) [reply code/modules/atmospherics/machinery/portable/canister.dm]: its for holding the bomb and not transferring gas between them and showing it to people

**Comment by iloveloopers** [comment]
oops nevermind didn't realize it was only when broken

File: code/modules/atmospherics/machinery/portable/canister.dm (hunk 2 above)
Comment (iamthesilly) [reply code/modules/atmospherics/machinery/portable/canister.dm]: It already holds the tank whether or not you open the canister to transfer gas.

Comment (void5402) [reply code/modules/atmospherics/machinery/portable/canister.dm]: will this do?

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 3]
```diff
@@ -310,12 +312,16 @@
 		. += "can-open"
 	if(holding)
 		. += "can-tank"
+		if(holding.bomb_status)
```
Comment (iamthesilly) [inline code/modules/atmospherics/machinery/portable/canister.dm]: Why is this here twice?

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 4]
```diff
@@ -310,12 +312,16 @@
...
+			. += holding
 	if(connected_port)
 		. += "can-connector"
 
 	var/light_state = get_pressure_state(air_contents?.return_pressure())
 	if(light_state) //happens when pressure is below 10kpa which means no light
 		. += mutable_appearance(icon, light_state)
+	if(holding.bomb_status)
```
Comment (iamthesilly) [inline code/modules/atmospherics/machinery/portable/canister.dm]: This needs a null check so it doesn't cause runtime errors.
```suggestion
	if(holding?.bomb_status)
```

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Review by SapphicOverload — APPROVED** [review]

**Comment by MajManatee** [comment]
pov you are trying to do fusion in a can and accidentally depopulate the station

**Comment by Moltijoe** [comment]
Balance Council:
>What? this just lets atmos and toxins create more maxcaps

**Comment by iloveloopers** [comment]
you can make and store singletank bombs in a bag which are capable of gibbing, but a huge canister that everyone can see is what crosses the line

**Comment by neeshacark** [comment]
i mean hot take maybe but atmos can do much worse than a maxcap - floods of any kind are WAY more damaging.

and toxins can already make maxcaps so idk why letting them do it in a slightly different way is bad

**Comment by Moltijoe** [comment]
> i mean hot take maybe but atmos can do much worse than a maxcap - floods of any kind are WAY more damaging.
> 
> and toxins can already make maxcaps so idk why letting them do it in a slightly different way is bad

the thing is, scientists are already antifun with maxcap bombing, but they have limited ttv, so eventually the bombs run out
with canister bombs, there's far more canisters, so they're not likely to ever run out
also, it only relying on pressure instead of a proper gas mixture doesn't exactly help it's case either

When it comes to floods, they're annoying, but it's damage is SO widespread that only hijack/martyr engineers can do it without getting bwoinked
Bombs are significantly more controlled, meaning even civilians can get away with using them under certain circumstances

**Comment by Hislyna** [comment]
You should change it so that if its a too high pressure threshold, theres a chance that itll just blow up in their face
